### PR TITLE
(torchx/schedulers) AWSBatchScheduler; add privileged option to enabl…

### PR DESCRIPTION
**Pull request by kiukchung**
1. Adds privileged runopt to aws batch scheduler
2. Also fixes a double call to create trackers from env in `tracker/api.py`, which discards parent run source information since a new set of trackers are created and hooked to the AppRun.

Test plan:

Added unittests to cover relevant changes

### CODE REVIEW

File: torchx/tracker/api.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def run_from_env() -> AppRun:
             for tracker in trackers:
                 tracker.add_source(torchx_job_id, parent_run_id, artifact_name=None)
 
-        return AppRun(id=torchx_job_id, backends=trackers_from_environ())
+        return AppRun(id=torchx_job_id, backends=trackers)
```
Comment (d4l3k) [inline torchx/tracker/api.py:284]: This should really be in a different PR -- though it's small enough

**Review by d4l3k — APPROVED** [review]
LGTM

**Comment by d4l3k** [comment]
There's a pyre failure

File: torchx/tracker/api.py (hunk 1 above)
Comment (kiukchung) [reply torchx/tracker/api.py:284]: yea... you are right, I should've done this in a different PR... just got a bit lazy at 3am... will make sure I split the PRs going forward.

File: torchx/tracker/test/api_test.py [hunk 2]
```diff
@@ -119,9 +120,18 @@ def test_env_factory_test(self) -> None:
             "torchx.tracker.api.load_group",
             return_value={"tracker1": tracker_factory},
         ):
-            app_run = tracker.app_run_from_env()
+            app_run = app_run_from_env()
             self.assertEqual(app_run.id, self.run_id)
-            self.assertEqual(TestTrackerBackend, type(app_run.backends[0]))
+            self.assertEqual(1, len(app_run.backends))
```
Comment (d4l3k) [inline torchx/tracker/test/api_test.py:125]: likely can just change backends type to be Sized instead of Iterable

Comment (kiukchung) [reply torchx/tracker/test/api_test.py]: yep just fixed.
